File: dehaze-python/app/models/entity/sys_dept.py
"""
部门实体模型
"""

from sqlalchemy import BigInteger, Integer, String
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysDept(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_dept"
    __table_args__ = {"comment": "部门表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    name: Mapped[str] = mapped_column(String(64), nullable=False, default="", comment="部门名称")
    parent_id: Mapped[int] = mapped_column(
        BigInteger, nullable=False, default=0, comment="父节点id"
    )
    tree_path: Mapped[str] = mapped_column(String(255), default="", comment="父节点id路径")
    sort: Mapped[int] = mapped_column(Integer, default=0, comment="显示顺序")
    status: Mapped[int] = mapped_column(
        mysql_types.TINYINT, nullable=False, default=1, comment="状态(1:正常;0:禁用)"
    )

File: dehaze-go/internal/model/sys_dept.go
package model

// SysDept 部门表
type SysDept struct {
	BaseModel
	Name     string `gorm:"column:name;type:varchar(64);not null;default:'';comment:部门名称" json:"name"`
	ParentID int64  `gorm:"column:parent_id;type:bigint;not null;default:0;comment:父节点id" json:"parentId"`
	TreePath string `gorm:"column:tree_path;type:varchar(255);default:'';comment:父节点id路径" json:"treePath"`
	Sort     int    `gorm:"column:sort;type:int;default:0;comment:显示顺序" json:"sort"`
	Status   int8   `gorm:"column:status;type:tinyint;not null;comment:状态(1:正常,0:禁用)" json:"status"`
	Deleted  int64  `gorm:"column:deleted;type:bigint;default:0;comment:逻辑删除标识(0:未删除;>0:已删除,值为删除时的行id)" json:"deleted"`
}
